File: Trabalho Metodos de Ordenacao - Aula 29.08/principal.py
import threading
from util import Util
from ordenacao import Ordenacao

# Criar listas de alunos
alunos_bolha = []
alunos_selecao = []
alunos_insercao = []
alunos_sort = []

# Gerar 60000 alunos em cada lista (reduzir para 1000 para testes)
Util.gerar_alunos(alunos_bolha, 10000)
alunos_selecao = alunos_bolha.copy()
alunos_insercao = alunos_bolha.copy()
alunos_sort = alunos_bolha.copy()
# Copia a lista alunos_bolha para as outras listas, para garantir que cada lista comece com os mesmos dados.

# Funções para ordenar usando diferentes métodos
def ordenar_bolha():
    Ordenacao.bolha(alunos_bolha)
    print("Ordenacao por Bolha concluida.")

def ordenar_selecao():
    Ordenacao.selecao(alunos_selecao)
    print("Ordenacao por Selecao concluida.")

def ordenar_insercao():
    Ordenacao.insercao(alunos_insercao)
    print("Ordenacao por Insercao concluida.")

def ordenar_sort():
    alunos_sort.sort(key=lambda aluno: (aluno.nome, aluno.idade))
    print("Ordenacao com sort do Python concluida.")

# Criação e execução das threads para ordenação

threads = [
    threading.Thread(target=ordenar_bolha),
    threading.Thread(target=ordenar_selecao),
    threading.Thread(target=ordenar_insercao),
    threading.Thread(target=ordenar_sort)
]


# Iniciar as threads
for thread in threads:
    thread.start()

# Aguardar o término de todas as threads para prosseguir
for thread in threads:
    thread.join()

File: Trabalho Metodos de Ordenacao - Aula 29.08/Metodos de Ordenacao Java - Aula 05.09/src/metodosdeordenação/Principal.java
import java.util.ArrayList;
import java.util.List;

public class Principal {
    public static void main(String[] args) {
        // Criação das listas de alunos
        List<Aluno> alunosBolha = Util.gerarAlunos(10000);  // 10000 para um teste mais eficiente
        List<Aluno> alunosSelecao = new ArrayList<>(alunosBolha);
        List<Aluno> alunosInsercao = new ArrayList<>(alunosBolha);
        List<Aluno> alunosSort = new ArrayList<>(alunosBolha);

        // Funções para ordenar usando os métodos
        Runnable ordenarBolha = () -> {
            Ordenacao.bolha(alunosBolha);
            System.out.println("Ordenação por Bolha concluída.");
        };

        Runnable ordenarSelecao = () -> {
            Ordenacao.selecao(alunosSelecao);
            System.out.println("Ordenação por Seleção concluída.");
        };

        Runnable ordenarInsercao = () -> {
            Ordenacao.insercao(alunosInsercao);
            System.out.println("Ordenação por Inserção concluída.");
        };

        Runnable ordenarSort = () -> {
            alunosSort.sort((a1, a2) -> {
                int compare = a1.getNome().compareTo(a2.getNome());
                return compare == 0 ? Integer.compare(a1.getIdade(), a2.getIdade()) : compare;
            });
            System.out.println("Ordenação com sort do Java concluída.");
        };

        // criando threads para ordenação
        Thread threadBolha = new Thread(ordenarBolha);
        Thread threadSelecao = new Thread(ordenarSelecao);
        Thread threadInsercao = new Thread(ordenarInsercao);
        Thread threadSort = new Thread(ordenarSort);

        // Início das threads
        threadBolha.start();
        threadSelecao.start();
        threadInsercao.start();
        threadSort.start();

        // Espera terminar todas as threads
        try {
            threadBolha.join();
            threadSelecao.join();
            threadInsercao.join();
            threadSort.join();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }

        // Exibindo os 10 primeiros alunos de cada lista ordenada
        Util.exibir(alunosSort.subList(0, 10), "Exibindo os 10 primeiros alunos ordenados (sort Java):");
    }
}
